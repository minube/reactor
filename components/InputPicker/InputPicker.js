import {
  arrayOf, bool, func, string,
} from 'prop-types';
import React, { PureComponent } from 'react';
import { Picker, View } from 'react-native';

import { InputLabel } from '../Input';
import Text from '../Text';
import styles from './InputPicker.style';

class InputPicker extends PureComponent {
  static propTypes = {
    dataSource: arrayOf(string),
    disabled: bool,
    error: string,
    hint: string,
    label: string,
    onChange: func,
  };

  static defaultProps = {
    dataSource: [],
    disabled: false,
    error: undefined,
    hint: undefined,
    label: undefined,
    onChange: undefined,
  };

  render() {
    const {
      props: {
        dataSource = [], disabled, error, hint, label, onChange, ...inherit
      },
    } = this;

    return (
      <View style={[styles.container, inherit.style]}>
        { (label || (!disabled && error)) && <InputLabel value={label} error={error} /> }
        <View
          style={[
            styles.input,
            !disabled && error && styles.inputError,
            disabled && styles.inputDisabled,
          ]}
        >
          <Picker
            mode="dropdown"
            {...inherit}
            enabled={!disabled}
            onValueChange={onChange}
            selectedValue={inherit.value}
            style={[styles.picker, disabled && styles.pickerDisabled]}
          >
            { dataSource.map(item => <Picker.Item key={item} label={item} value={item} />)}
          </Picker>
        </View>
        { hint && (
          <Text caption level={2} lighten>
            {hint}
          </Text>)}
      </View>
    );
  }
}

export default InputPicker;
