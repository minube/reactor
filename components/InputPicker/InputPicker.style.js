import { StyleSheet } from 'react-native';

import { THEME } from '../../common';

const {
  COLOR, INPUT_HEIGHT, OFFSET, UNIT,
} = THEME;

export default StyleSheet.create({
  container: {
    minHeight: INPUT_HEIGHT,
    marginBottom: OFFSET,
  },

  label: {
    marginVertical: UNIT / 4,
  },

  input: {
    backgroundColor: COLOR.WHITE,
    paddingRight: UNIT * 0.5,
    borderColor: COLOR.BORDER,
    borderWidth: 1,
    borderRadius: UNIT / 4,
    width: '100%',
  },

  inputError: {
    borderColor: COLOR.ERROR,
  },

  inputFocus: {
    borderColor: COLOR.PRIMARY,
  },

  inputDisabled: {
    backgroundColor: COLOR.BACKGROUND,
  },

  picker: {
    backgroundColor: COLOR.TRANSPARENT,
    borderWidth: 0,
    height: UNIT * 3.3,
    color: COLOR.TEXT,
    // @TODO: How we can set size and weight
    // fontSize: FONT.SIZE.REGULAR,
    // fontWeight: FONT.WEIGHT.REGULAR,
    // ...Platform.select({
    //   web: {
    //     fontFamily: FONT.FAMILY,
    //   },
    // }),
  },

  pickerDisabled: {
    color: COLOR.BASE,
  },

  active: {
    color: COLOR.ACCENT,
  },
});
