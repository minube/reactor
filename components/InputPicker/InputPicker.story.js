import React from 'react';
import { action } from '@storybook/addon-actions';
import { storiesOf } from '@storybook/react';
import { text, boolean, number, select } from '@storybook/addon-knobs/react';

import InputPicker from './InputPicker';

const DATASOURCE = ['one', 'two', 'three', 'four', 'five'];

storiesOf('🛠 InputPicker', module)
  .addWithJSX('default', () => (
    <InputPicker />
  ))
  .addWithJSX('datasource', () => (
    <InputPicker datasource dataSource={DATASOURCE} />
  ))
  .addWithJSX('disabled', () => (
    <InputPicker disabled dataSource={DATASOURCE} />
  ))
  .addWithJSX('error', () => (
    <InputPicker error="Required field" dataSource={DATASOURCE} />
  ))
  .addWithJSX('hint', () => (
    <InputPicker hint="Optional field" dataSource={DATASOURCE} />
  ))
  .addWithJSX('label', () => (
    <InputPicker label="Username" dataSource={DATASOURCE} />
  ))
  .addWithJSX('value', () => (
    <InputPicker value="three" dataSource={DATASOURCE} />
  ))
  .addWithJSX('⚡ onChange', () => (
    <InputPicker onChange={action('InputPicker.onChange()')} title="Press me" dataSource={DATASOURCE} />
  ))
  .addWithJSX('style', () => (
    <InputPicker style={{ backgroundColor: 'orange', padding: 10, width: 256 }} dataSource={DATASOURCE} />
  ))
  .addWithJSX('🏀 Playground', () => (
    <InputPicker
      disabled={boolean('disabled', false)}
      error={text('error', null)}
      hint={text('hint', null)}
      label={text('label', null)}
    />
  ))

