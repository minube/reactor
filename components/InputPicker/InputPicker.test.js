import React from 'react';
import renderer from 'react-test-renderer';

import InputPicker from './InputPicker';

describe('<InputPicker>', () => {
  it('renders', () => {
    const tree = renderer.create(<InputPicker />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {disabled}', () => {
    const tree = renderer.create(<InputPicker disabled />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {error}', () => {
    const tree = renderer.create(<InputPicker error="Hello World" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {hint}', () => {
    const tree = renderer.create(<InputPicker hint="Hello World" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {label}', () => {
    const tree = renderer.create(<InputPicker label="Hello World" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {value}', () => {
    const tree = renderer.create(<InputPicker value="three" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {style}', () => {
    const tree = renderer.create(<InputPicker style={{ backgroundColor: 'orange', padding: 10, width: 256 }} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

