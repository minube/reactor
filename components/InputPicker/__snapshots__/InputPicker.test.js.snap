// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputPicker> renders 1`] = `
<View
  style={
    Array [
      Object {
        "marginBottom": 16,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#dbdbdb",
          "borderRadius": 2.5,
          "borderWidth": 1,
          "paddingRight": 5,
          "width": "100%",
        },
        undefined,
        false,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
            "height": 33,
          },
          false,
        ]
      }
    >
      <RCTPicker
        items={Array []}
        onChange={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        selectedIndex={0}
        style={
          Array [
            Object {
              "height": 216,
            },
            undefined,
          ]
        }
      />
    </View>
  </View>
</View>
`;

exports[`<InputPicker> when {disabled} 1`] = `
<View
  style={
    Array [
      Object {
        "marginBottom": 16,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#dbdbdb",
          "borderRadius": 2.5,
          "borderWidth": 1,
          "paddingRight": 5,
          "width": "100%",
        },
        false,
        Object {
          "backgroundColor": "#f4f4f4",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
            "height": 33,
          },
          undefined,
        ]
      }
    >
      <RCTPicker
        items={Array []}
        onChange={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        selectedIndex={0}
        style={
          Array [
            Object {
              "height": 216,
            },
            undefined,
          ]
        }
      />
    </View>
  </View>
</View>
`;

exports[`<InputPicker> when {error} 1`] = `
<View
  style={
    Array [
      Object {
        "marginBottom": 16,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#dbdbdb",
          "borderRadius": 2.5,
          "borderWidth": 1,
          "paddingRight": 5,
          "width": "100%",
        },
        Object {
          "borderColor": "#ef5160",
        },
        false,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
            "height": 33,
          },
          false,
        ]
      }
    >
      <RCTPicker
        items={Array []}
        onChange={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        selectedIndex={0}
        style={
          Array [
            Object {
              "height": 216,
            },
            undefined,
          ]
        }
      />
    </View>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#4a4a4a",
          "fontFamily": undefined,
        },
        false,
        false,
        false,
        false,
        Object {
          "fontSize": 16,
          "lineHeight": 20.8,
        },
        Object {
          "fontSize": 12,
          "lineHeight": 14.399999999999999,
        },
        false,
        false,
        false,
        false,
        Object {
          "color": "#9b9b9b",
        },
        false,
        false,
        Object {
          "color": "#ef5160",
          "marginVertical": 2.5,
        },
      ]
    }
  >
    Hello World
  </Text>
</View>
`;

exports[`<InputPicker> when {hint} 1`] = `
<View
  style={
    Array [
      Object {
        "marginBottom": 16,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#dbdbdb",
          "borderRadius": 2.5,
          "borderWidth": 1,
          "paddingRight": 5,
          "width": "100%",
        },
        undefined,
        false,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
            "height": 33,
          },
          false,
        ]
      }
    >
      <RCTPicker
        items={Array []}
        onChange={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        selectedIndex={0}
        style={
          Array [
            Object {
              "height": 216,
            },
            undefined,
          ]
        }
      />
    </View>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#4a4a4a",
          "fontFamily": undefined,
        },
        false,
        false,
        false,
        false,
        Object {
          "fontSize": 16,
          "lineHeight": 20.8,
        },
        Object {
          "fontSize": 12,
          "lineHeight": 14.399999999999999,
        },
        false,
        false,
        false,
        false,
        Object {
          "color": "#9b9b9b",
        },
        false,
        false,
        Object {
          "marginVertical": 2.5,
        },
      ]
    }
  >
    Hello World
  </Text>
</View>
`;

exports[`<InputPicker> when {label} 1`] = `
<View
  style={
    Array [
      Object {
        "marginBottom": 16,
      },
      undefined,
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#4a4a4a",
          "fontFamily": undefined,
        },
        false,
        false,
        false,
        false,
        Object {
          "fontSize": 16,
          "lineHeight": 20.8,
        },
        Object {
          "fontSize": 12,
          "lineHeight": 14.399999999999999,
        },
        false,
        false,
        false,
        false,
        Object {
          "color": "#9b9b9b",
        },
        false,
        false,
        Object {
          "marginVertical": 2.5,
        },
      ]
    }
  >
    Hello World
  </Text>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#dbdbdb",
          "borderRadius": 2.5,
          "borderWidth": 1,
          "paddingRight": 5,
          "width": "100%",
        },
        undefined,
        false,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
            "height": 33,
          },
          false,
        ]
      }
    >
      <RCTPicker
        items={Array []}
        onChange={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        selectedIndex={0}
        style={
          Array [
            Object {
              "height": 216,
            },
            undefined,
          ]
        }
      />
    </View>
  </View>
</View>
`;

exports[`<InputPicker> when {style} 1`] = `
<View
  style={
    Array [
      Object {
        "marginBottom": 16,
      },
      Object {
        "backgroundColor": "rgba(0,255,0,0.25)",
        "padding": 10,
        "width": 256,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#dbdbdb",
          "borderRadius": 2.5,
          "borderWidth": 1,
          "paddingRight": 5,
          "width": "100%",
        },
        undefined,
        false,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
            "height": 33,
          },
          false,
        ]
      }
    >
      <RCTPicker
        items={Array []}
        onChange={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        selectedIndex={0}
        style={
          Array [
            Object {
              "height": 216,
            },
            undefined,
          ]
        }
      />
    </View>
  </View>
</View>
`;

exports[`<InputPicker> when {value} 1`] = `
<View
  style={
    Array [
      Object {
        "marginBottom": 16,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#dbdbdb",
          "borderRadius": 2.5,
          "borderWidth": 1,
          "paddingRight": 5,
          "width": "100%",
        },
        undefined,
        false,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
            "height": 33,
          },
          false,
        ]
      }
    >
      <RCTPicker
        items={Array []}
        onChange={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        selectedIndex={0}
        style={
          Array [
            Object {
              "height": 216,
            },
            undefined,
          ]
        }
      />
    </View>
  </View>
</View>
`;
